Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -141,18 +141,17 @@ void sendOnChannel(FairMQDevice& device, FairMQMessagePtr&& headerMessage, FairM
   sendOnChannel(device, out, spec, channelRetriever);
 }
 
-InjectorFunction o2DataModelAdaptor(OutputSpec const& spec, uint64_t startTime, uint64_t step)
+InjectorFunction o2DataModelAdaptor(OutputSpec const& spec, uint64_t step)
 {
-  auto timesliceId = std::make_shared<size_t>(startTime);
-  return [timesliceId, step, spec](FairMQDevice& device, FairMQParts& parts, ChannelRetriever channelRetriever) {
+  return [step, spec](FairMQDevice& device, FairMQParts& parts, ChannelRetriever channelRetriever) {
     for (size_t i = 0; i < parts.Size() / 2; ++i) {
       auto dh = o2::header::get<DataHeader*>(parts.At(i * 2)->GetData());
 
-      DataProcessingHeader dph{*timesliceId, 0};
+      // Now that the first orbit is part of the DataHeader, we can use that if
+      // the DataHeader is present.
+      DataProcessingHeader dph{dh->firstTForbit, 0};
```

Comment:
Not sure if I understand the comment, the DataHeader is always present, isn't it? We do not check for nullptr.

Reply:
DataHeader might not be present if you are injecting non-O2Messages in the flow. I.e. o2DPLAdapter cannot be use in that case. I will clarify.

Reply:
ok, then we need to catch this and throw an exception

Reply:
This will actually happen when you do the get<> no?

Reply:
no, returns nullptr if the header type is not on the stack. This is to support optional headers on the stack